extensions_admin/pulp_ostree/extensions/admin/cudl.py

```diff
@@ -49,15 +59,27 @@ def _describe_distributors(self, user_input):
         is needed to create distributor configs.
 
         :param user_input:  dictionary of data passed in by okaara
-        :type  user_inpus:  dict
+        :type  user_input:  dict
 
         :return:    list of dict containing distributor_type_id,
                     repo_plugin_config, auto_publish, and distributor_id (the same
                     that would be passed to the RepoDistributorAPI.create call).
         :rtype:     list of dict
         """
-        config = {}
-        auto_publish = user_input.get('auto-publish', True)
+        relative_path = user_input.get(OPT_RELATIVE_PATH.keyword)
+        auto_publish = user_input.get(OPT_AUTO_PUBLISH.keyword, True)
+
+        # relative path derived using the path component of the feel url when not specified
+        if not relative_path:
+            feed_url = user_input.get(self.options_bundle.opt_feed.keyword)
+            if feed_url:
+                url = urlparse(feed_url)
+                relative_path = url.path
```

Comment:
On 03/18/2015 05:04 PM, Jeff Ortel wrote:
> The distributor does not know anything about the |feed url|. that is an
> importer property.

If we don't solve this server side, isn't the bug I pointed out going to
happen? A user sets a relative path, later changes a feed, and suddenly
the repo is published somewhere else? That seems bad. Is there another
solution that doesn't depend on it being done client-side?

-- 
Randy Barlow